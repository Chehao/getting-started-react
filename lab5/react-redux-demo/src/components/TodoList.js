import PropTypes from 'prop-types';
import React from 'react';
import Todo from './Todo';
const TodoList = ({ todos, toggleTodo }) => (
  <ul>
    {todos.map((todo, index) => (
      <Todo key={index} {...todo} onClick={() => toggleTodo(index)} />
    ))}
  </ul>
)

TodoList.propTypes = {
  todos: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      completed: PropTypes.bool.isRequired,
      text: PropTypes.string.isRequired
    }).isRequired
  ).isRequired,
  toggleTodo: PropTypes.func.isRequired
}

export default TodoList