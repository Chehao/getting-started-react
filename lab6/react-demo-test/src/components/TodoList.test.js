import React from 'react';
import ReactDOM from 'react-dom';
import TodoList from './TodoList';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Todo from './Todo';
import 'jest-enzyme';

it('1. render TodoList without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TodoList todos={[]} toggleTodo={() => {}} />, div);
});

it('2. render empty TodoList', () => {
  const tree = renderer.create(<TodoList todos={[]} toggleTodo={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('3. render homework TodoList', () => {
  
  const tree = renderer
    .create(
      <TodoList
        todos={[
          {
            id: 6,
            completed: false,
            text: 'homework2'
          },{
            id: 7,
            completed: false,
            text: 'homework3'
          }
        ]}
        toggleTodo={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('4. shallow render TodoList should contains homework with id 7', ()=>{
  const wrapper = shallow(<TodoList todos={[
    {
      id: 7,
      completed: false,
      text: 'homework'
    }
  ]} toggleTodo={() => {}} />);
  const homework = <Todo id={7} completed={false} onClick={() => {}} text="homework"/>
  expect(wrapper.find(Todo)).toHaveLength(1);
  expect(wrapper.find(Todo)).toHaveProp('text', 'homework')
  expect(wrapper.find(Todo)).toHaveProp('id', 7)
  
})