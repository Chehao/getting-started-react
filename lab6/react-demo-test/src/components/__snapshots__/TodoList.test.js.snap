// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2. render empty TodoList 1`] = `<ul />`;

exports[`3. render homework TodoList 1`] = `
<ul>
  <li
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    homework2
  </li>
  <li
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    homework3
  </li>
</ul>
`;
